vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,478 @@
-likely increase the variance of the model, and therefore, may lead to less
-accurate forecasts for the variable of interest.
+Because of the flexibility of `{parsnip}`, there are a whole host of models
+available to us[^5]; as an example, we could have just as easily substituted a
+non-linear random forest model from `{ranger}`:
 
-Another property of the basic model is the predictive interval. We describe this
-in more detail in a different vignette, but it is easy to request multiple
-quantiles.
-
-```{r differential-levels}
-out_q <- arx_forecaster(jhu, "death_rate", c("case_rate", "death_rate"),
+```{r rand_forest_ex, warning=FALSE}
+two_week_ahead <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
+  outcome = "death_rate",
+  trainer = rand_forest(mode = "regression"),
+  predictors = c("death_rate"),
   args_list = arx_args_list(
-    quantile_levels = c(.01, .025, 1:19 / 20, .975, .99)
+    lags = list(c(0, 7, 14)),
+    ahead = 14
   )
 )
 ```
 
-The column `.pred_dstn` in the `predictions` object is actually a "distribution"
-here parameterized by its quantiles. For this default forecaster, these are
-created using the quantiles of the residuals of the predictive model (possibly
-symmetrized). Here, we used 23 quantiles, but one can grab a particular
-quantile,
+Any other customization is routed through `arx_args_list()`; for example, if we
+wanted to increase the number of quantiles fit:
 
-```{r q1}
-round(head(quantile(out_q$predictions$.pred_distn, p = .4)), 3)
+```{r make-quantile-levels-forecasts, warning=FALSE}
+two_week_ahead <- arx_forecaster(
+  covid_case_death_rates |>
+    filter(time_value <= forecast_date, geo_value %in% used_locations),
+  outcome = "death_rate",
+  trainer = quantile_reg(),
+  predictors = c("death_rate"),
+  args_list = arx_args_list(
+    lags = list(c(0, 7, 14)),
+    ahead = 14,
+    ############ changing quantile_levels ############
+    quantile_levels = c(0.05, 0.1, 0.2, 0.3, 0.5, 0.7, 0.8, 0.9, 0.95)
+    ##################################################
+  )
+)
+hardhat::extract_fit_engine(two_week_ahead$epi_workflow)
 ```
 
-or extract the entire distribution into a "long" `epi_df` with `quantile_levels`
-being the probability and `values` being the value associated to that quantile.
-
-```{r q2}
-out_q$predictions %>%
-  pivot_quantiles_longer(.pred_distn)
+See the function documentation of `arx_args_list()` for more examples of the modifications available.
+If you are looking for even further modifications, you will want a custom
+workflow, for which you should see the [guts
+vignette](preprocessing-and-models).
+
+## Generating multiple aheads
+Frequently, one doesn't want just a forecast for a single day, but a trajectory
+of forecasts for several weeks.
+The way to do this using `arx_forecaster()` is by looping over aheads; for
+example, to predict every day over 4 weeks:
+
+```{r temp-thing}
+all_canned_results <- lapply(
```

Comment:
suggestion: it's a little odd to have to loop over aheads "manually". Seems like we could easily modify `arx_forecaster` to map over a list of aheads if length > 1. Same with autoplot.

Reply:
Fair point for `arx_forecaster`. @dajmcdon thoughts on adding a loop to `arx_forecaster`?

I believe autoplot already does this

Reply:
The type stability on its own seems like a good argument for the current version. The extra overhead of the user doing a map isn't that great, and we're demoing how to do it here.